prov/gni/src/gnix_rma.c

```diff
@@ -140,13 +184,15 @@ static int __gnix_rma_post_err(struct gnix_tx_descriptor *txd)
 
 	req->tx_failures++;
 	if (req->tx_failures < req->gnix_ep->domain->params.max_retransmits) {
+		_gnix_nic_tx_free(req->gnix_ep->nic, txd);
+
```

Comment:
Was this a bug or is there something about handling unaligned accesses that makes this necessary?